input/drivers_joypad/udev_joypad.c

```diff
@@ -147,6 +151,37 @@ static int udev_open_joystick(const char *path)
    return -1;
 }
 
+static bool udev_set_rumble_gain(unsigned i, unsigned gain)
+{
+   struct input_event ie;
+   struct udev_joypad *pad = (struct udev_joypad*)&udev_pads[i];
+
+   if (pad->fd < 0)
+      return false;
+
+   if (pad->rumble_gain == gain)
+      return true;
+
+   /* Does not support > 100 gains */
+   if (gain > 100)
+      return false;
```

Comment:
@davidgfnet I was wondering, is this correct? Checking the gain against the upper bound _after_ attempting to apply it?

Reply:
I think that looks okay - it's still performing the check before accessing the hardware. Could combine the checks, though, I suppose - i.e.:

```c
   struct input_event ie;
   struct udev_joypad *pad = (struct udev_joypad*)&udev_pads[i];

   /* Does not support > 100 gains */
   if ((pad->fd < 0) ||
       (gain > 100))
      return false;

   if (pad->rumble_gain == gain)
      return true;
```